### PR TITLE
Using directives for input validation. Replaced the messages with toa…

### PR DESCRIPTION
**Pull request by jeroenm2**
…str messages. Fixed a small thing in the backend

### CODE REVIEW

File: public/javascripts/controllers.js [hunk 1]
```diff
@@ -795,43 +797,14 @@ angular.module('uforia')
...
-                $scope.cuErrorMessages.push('Make sure the passwords match.');
-              }
-            }
-
-            if(skip === 0){
-              if(typeof user.password2 === "undefined"){
-                $scope.cuErrorMessages.push('Make sure the passwords match.');
-              }
-            }
+          var skip = 0;
```
Comment (Moduo) [inline public/javascripts/controllers.js]: Is `skip` still used? I don't see any checks on it. Could it be removed?

File: public/javascripts/controllers.js [hunk 2]
```diff
@@ -843,14 +816,13 @@ angular.module('uforia')
                   console.log(user);
                   $http.post('/api/edit_user', $scope.editUser)
                       .success(function (data) {
+                            console.log(data);
```
Comment (Moduo) [inline public/javascripts/controllers.js]: Do we still need this logging?

File: public/javascripts/controllers.js [hunk 3]
```diff
@@ -888,82 +857,50 @@ angular.module('uforia')
...
-                      );
-                }
-
-              }else{
-                $scope.cuErrorMessages.push('Unknown error.');
+            
+            var addUser = true;
+  
+            // Check if user exists
+            angular.forEach(users, function(value, key){
```
Comment (Moduo) [inline public/javascripts/controllers.js]: This seems like a task for the backend. Not the frontend.

File: public/javascripts/controllers.js [hunk 4]
```diff
@@ -888,82 +857,50 @@ angular.module('uforia')
...
+                            });
+                            $scope.searchCollection = $scope.rowCollection;
+                          }
+                        }
+                        // Close modal
+                        $scope.modalInstance.dismiss();
+                      }
+                  );
+            } else if (!addUser){
+              toastr.error('An user with this email address already exists');
```
Comment (Moduo) [inline public/javascripts/controllers.js]: It's 

> A user
not 

> An user

File: routes/api.js [hunk 5]
```diff
@@ -865,11 +865,12 @@ router.post('/edit_user', function(req, res){
   var id = user.id;
   delete user.id;
 
-  if (user.password != 'undefined') {
-    bcrypt.hash(user['password'], null, null, function(err, hash){
+  bcrypt.hash(user['password'], null, null, function(err, hash){
+    var hash = hash;
```
Comment (Moduo) [inline routes/api.js]: What are you trying to do with this?

File: public/javascripts/controllers.js (hunk 3 above)
Comment (jeroenm2) [inline public/javascripts/controllers.js]: Is on the todo list for the sprint, for now this works

File: routes/api.js (hunk 5 above)
Comment (jeroenm2) [inline routes/api.js]: Prevent updating a password if the password is unchanged

File: public/javascripts/controllers.js (hunk 3 above)
Comment (Moduo) [inline public/javascripts/controllers.js]: Because we're considering in using MySQL for the users, we indeed postpone this to a later stadium.

**Pull request merged**